File: Server/Csharp_Server/UserServer/Utils/UserName.cs
using PGAdminDAL;
using System;
using System.Text;

namespace UserServer.utils
{
    public class UserName
    {
        private static Random random = new Random();

        internal List<string> GenerateAdditionalNicknames(string nickname, AppDbContext context)
        {
            List<string> additionalNicknames = new List<string>();

            additionalNicknames.Add(GenerateRandomNickname(nickname, context));
            additionalNicknames.Add(GenerateRandomNickname(nickname, context));
            additionalNicknames.Add(GenerateRandomNickname(nickname, context));
            additionalNicknames.Add(GenerateRandomNickname(nickname, context));
            additionalNicknames.Add(GenerateRandomNickname(nickname, context));
            additionalNicknames.Add(GenerateRandomNickname(nickname, context));
            additionalNicknames.Add(GenerateRandomNickname(nickname, context));

            return additionalNicknames;
        }

        private string GenerateRandomNickname(string nick, AppDbContext context)
        {
            const string chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
            StringBuilder nickname = new StringBuilder(nick);
            string newNickname;

            if (random.Next(0, 2) == 1)
            {
                nickname.Append('_');
            }

            do
            {
                nickname.Clear();
                nickname.Append(nick);
                int randomLength = random.Next(0, 6);
                for (int i = 0; i < randomLength; i++)
                {
                    nickname.Append(chars[random.Next(chars.Length)]);
                }
                newNickname = nickname.ToString();
            }
            while (context.Users.Any(u => u.UserName == newNickname));


            return nickname.ToString();
        }
    }
}

File: Server/Spring_Server/src/main/java/com/Spacegram/server/utility/UserServices/userName.java
package com.Spacegram.server.utility.UserServices;

import com.Spacegram.server.PostgresDB.Models.AspNetUsers;
import com.Spacegram.server.utility.Sql.Users;
import org.hibernate.Transaction;

import org.hibernate.Session;
import java.util.ArrayList;
import java.util.Random;

public class userName
{
    Random random = new Random();
    public String GenerateRandomNickname(String nickname, Session users, ArrayList LastNames)
    {
        final String chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
        StringBuilder sb = new StringBuilder(nickname);
        String newNickname;

        if(random.nextInt(0, 2) == 1){
            sb.append('_');
        }

        do
        {
            sb.delete(0, sb.length());
            sb.append(nickname);
            int randomLength = random.nextInt(0, 6);
            for (int i = 0; i < randomLength; i++)
            {
                sb.append(chars.charAt(random.nextInt(chars.length())));
            }
            newNickname = sb.toString();
        }
        while (LastNames.contains(newNickname) || Users.isNicknameExistInDatabase(newNickname, users));

        return newNickname;
    }




}
